paging/paging-common/src/main/kotlin/androidx/paging/PagingDataDiffer.kt

```diff
@@ -139,10 +140,24 @@ public abstract class PagingDataDiffer<T : Any>(
 
     public suspend fun collectFrom(pagingData: PagingData<T>) {
         collectFromRunner.runInIsolation {
-            receiver = pagingData.receiver
-
+            statelessReceiver = pagingData.receiver
```

Comment:
Since we still need to set these independently, let's just split them into two interfaces right away. Let's stick to `UiReceiver` and `HintReceiver` - those are good names. I'd prefer to avoid merging the half-way solution if we are already touching this part.

Reply:
done